### PR TITLE
Bug/ccls 2171 notification table and search formatting

### PR DESCRIPTION
**Pull request by farrell-m**
## Changes
* Fixed Notification search result display so that the results table fits on the page
* Fixed Notification search result sorting & pagination
* Fixed Notification search form to / from date display
* Fixed Notification search form date validation logic causing server error and form error display
* Upgraded Design Systems
  * GOVUK 5.0.0 -> 5.3.1
  * MoJ 2.0.0 -> 2.1.3
---

### Notification results before:
Screenshot 2024-05-09 at 14 31 23

### Notification results after:
Screenshot 2024-05-09 at 14 36 03

---

### Notification form before:
Screenshot 2024-05-09 at 14 32 30

### Notification form after:
Screenshot 2024-05-09 at 14 38 43

---

### Notification search form error before:
Screenshot 2024-05-09 at 14 34 02

### Notification search form error after:
Screenshot 2024-05-09 at 14 39 28

### CODE REVIEW

File: src/main/java/uk/gov/laa/ccms/caab/bean/NotificationSearchCriteria.java [hunk 1]
```diff
@@ -1,6 +1,8 @@
 package uk.gov.laa.ccms.caab.bean;
 
 import lombok.Data;
+import org.apache.commons.lang3.StringUtils;
```
Comment (PhilDigitalJustice) [inline src/main/java/uk/gov/laa/ccms/caab/bean/NotificationSearchCriteria.java:4]: import org.springframework.util.StringUtils; 

can we use this one instead?
might be worth removing all the apache ones as well, only see 6 references to it

File: src/main/java/uk/gov/laa/ccms/caab/bean/NotificationSearchCriteria.java [hunk 2]
```diff
@@ -97,21 +99,59 @@ public class NotificationSearchCriteria {
...
+   * Retrieves the formatted to date based on the day, month, and year values.
+   *
+   * @return The formatted to date (yyyy-MM-dd), or null if the date components are not valid
+   *         integers.
+   */
+  public String getDateTo() {
+    return getDate(notificationToDateYear, notificationToDateMonth, notificationToDateDay);
+  }
+
+  private String getDate(String yearInput, String monthInput, String dayInput) {
```
Comment (PhilDigitalJustice) [inline src/main/java/uk/gov/laa/ccms/caab/bean/NotificationSearchCriteria.java:122]: might be worth just adding a javadoc to this private method

File: src/main/java/uk/gov/laa/ccms/caab/bean/validators/notification/NotificationSearchValidator.java [hunk 3]
```diff
@@ -22,7 +24,7 @@
 @Slf4j
 public class NotificationSearchValidator extends AbstractValidator {
 
-  private static final String SOA_DATE_FORMAT = "dd/MM/yyyy";
+  private static final String ISO_DATE_FORMAT = "yyyy-MM-dd";
```
Comment (PhilDigitalJustice) [inline src/main/java/uk/gov/laa/ccms/caab/bean/validators/notification/NotificationSearchValidator.java:27]: I thought there was a reason we needed it as the SOA dateformat

File: src/main/java/uk/gov/laa/ccms/caab/constants/NotificationConstants.java
Comment (PhilDigitalJustice) [inline src/main/java/uk/gov/laa/ccms/caab/constants/NotificationConstants.java:1]: I know you didnt make this file but can you add java doc comments to each constant?

**Review by PhilDigitalJustice — CHANGES_REQUESTED** [review]
Few comments to address

File: src/main/java/uk/gov/laa/ccms/caab/bean/NotificationSearchCriteria.java (hunk 1 above)
Comment (farrell-m) [reply src/main/java/uk/gov/laa/ccms/caab/bean/NotificationSearchCriteria.java]: Switched all of them, in one case replaced with plain java (`StringUtils.repeat()` -> `str.repeat()`) as this wasn't provided by the Spring version. Also replaced some deprecated `isEmpty()`.

File: src/main/java/uk/gov/laa/ccms/caab/bean/NotificationSearchCriteria.java (hunk 2 above)
Comment (farrell-m) [reply src/main/java/uk/gov/laa/ccms/caab/bean/NotificationSearchCriteria.java:126]: Added

File: src/main/java/uk/gov/laa/ccms/caab/constants/NotificationConstants.java
Comment (farrell-m) [reply src/main/java/uk/gov/laa/ccms/caab/constants/NotificationConstants.java:1]: Added

**Review by PhilDigitalJustice — APPROVED** [review]
LGTM

**Pull request merged**